File: py-version/1-10/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        if(len(nums)<2):
            return False;
        buff_dict = {}
        for i in range(len(nums)):
            if nums[i] in buff_dict:
                return [buff_dict[nums[i]],i]
            else:
                buff_dict[target-nums[i]] = i

File: cpp-version/1-10/1-Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,vector<int>>mmp;
        for(int i=0; i<nums.size(); ++i) {
                mmp[nums[i]].push_back(i);
        }
        vector<int>ret;
        for(int i=0; i<nums.size(); ++i){
            int delta = target - nums[i];
            if(nums[i] == delta){
                if(mmp[nums[i]].size()>1){
                    ret.push_back(i);
                    ret.push_back(mmp[nums[i]][1]);
                    break;
                }
            }else if(mmp.find(delta)!=mmp.end()){
                ret.push_back(i);
                ret.push_back(mmp[delta][0]);
                break;
            }
        }
        return ret;
    }
};